homework/console/02/control.ts

```diff
@@ -0,0 +1,108 @@
+interface DoingInterface {
+  category: string;
+  content: string[];
+}
+interface MemoryInterface {
+  name: string;
+  gender: string;
+  age: number;
+  isStudent: boolean;
+  doing?: DoingInterface[];
+  hobby?: string[];
+}
+type ConvertedGender = { [key: string]: string };
+type actionType = MemoryInterface | string | number;
+
+const convertGender: ConvertedGender = {
+  female: '여성',
+  male: '남성',
+};
+
+const arr: number[] = [];
+const control = (type: string, action: actionType) => {
+  let gameStatus = '';
+  let result: string | number[] = '';
+  if (type === 'game') {
+    switch (action) {
+      case 'start':
+        gameStatus = '게임이 시작되었습니다!';
+        break;
+      case 'pause':
+        gameStatus = '게임이 중지 되었습니다!';
+        break;
+      case 'stop':
+        gameStatus = '게임이 종료 되었습니다!';
+        break;
+    }
+    result = gameStatus;
+  }
+
+  if (type === 'study') {
+    if (action > 0) {
+      !arr.includes(+action) && arr.push(+action);
+    } else {
+      !arr.includes(+action) && arr.pop();
+    }
```

Comment:
도움이 될지는 모르겠지만 아래와 같이 리팩토링 코드가 읽기 쉬워질것 같아요! 

```suggestion
const 스터디번호 = action;
const 스터디목록 = arr;

//이 부분은 하나의 기능만 하는 함수로 변경
const 스터디번호가정수 = 스터디번호 > 0;
const 스터디목록에이미추가됨 = 스터디목록.includes(+스터디번호);
const 스터디목록에추가된상태아님 = !스터디목록에이미추가됨;

if (스터디번호가정수) {
  스터디목록에추가된상태아님 && 스터디목록.push(+스터디번호);
} else {
  스터디목록에이미추가됨 && 스터디목록.pop();
}
```

Reply:
```
 const IS_POSITIVE_NUMBER = action > 0;
    const IN_STUDY = arr.includes(+action);
    const NOT_IN_STUDY = !IN_STUDY;

    if (IS_POSITIVE_NUMBER) {
      NOT_IN_STUDY && arr.push(+action);
    } else {
      NOT_IN_STUDY && arr.pop();
    }
    result = arr;
```

오 ... 한글변수명으로 써주시니 변수 선언 의도나 로직을 한번에 이해하기 정말 좋네용! 
감사합니당~~  이부분 수정해서 올릴게용 ! 👍🏻 💯